README.md

```diff
@@ -38,29 +38,115 @@ referenced to the user who requested it, and recall it when there's
 new a new push subscription update is left as an excerise for the
 reader.
 
-The data can be any serial content (string, bit array, serialized
+### Sending Data using One Call
+
+In many cases, you're code will probably be sending a single message
+to many recipients. To make things easier, there's a "One Call" option
+for WebPusher:
+
+```pythonstup
+    WebPusher(subscription_info,
+              data,
+              vapid_info)
+```
+This will encode `data`, add the appropriate VAPID headers created
+from the `vapid_info` and send it to the push server identified in the
+`subscription_info`.
+
+***data*** can be any serial content (string, bit array, serialized
 JSON, etc), but be sure that your receiving application is able to
 parse and understand it. (e.g. `data = "Mary had a little lamb."`)
 
-gcm_key is the API key obtained from the Google Developer Console.
-It is only needed if endpoint is
-https://android.googleapis.com/gcm/send
+***vapid_info*** is a JSON block that contains information about you, the
+sender. It is voluntary, however some Push Service providers may
+require it.)
 
-`headers` is a `dict`ionary of additional HTTP header values (e.g.
-[VAPID](https://github.com/mozilla-services/vapid/tree/master/python)
-self identification headers). It is optional and may be omitted.
+You may include `vapid_info` as a dictionary containing:
+```json
+{"private_key": "Private Key or File Path[1]",
+ "claims": {"sub": "mailto:YourEmailAddress"}}
+```
+
+The `private_key` may be either a base64 encoded DER formatted private
+key, or the path to an OpenSSL exported private key file.
 
-to send:
+e.g. the output of:
+```commandline
+openssl ecparam -name prime256v1 -genkey -noout -out private_key.pem
 ```
-WebPusher(subscription_info).send(data, headers)
+
+#### Example of One Call sends
+
+```pythonstub
+try:
+    WebPusher(
+        subscription_info={
+            "endpoint": "https://push.example.com/v1/12345",
+            "keys": {
+                "p256dh": "0123abcde...",
+                "auth": "abc123..."
+            }},
+        data="Mary had a little lamb, with a nice mint jelly",
+        vapid={
+            "private_key": "path/to/vapid_private.pem",
+            "claims": {
+                "sub": "YourNameHere@example.org",
+                "aud": "https://push.example.com/"
+            }
+        }
+    )
+except WebPushException as ex:
+    print("I'm sorry, Dave, but I can't do that: {}", repr(ex))
 ```
+
+### Methods
+
+If you expect to resend to the same recipient, or have more needs than just sending data quickly, you 
+can pass just `wp = WebPusher(subscription_info)`. This will return a `WebPusher` object.
```

Comment:
Ah, so the class instantiation may or may not result in sending data depending on the arguments... it'd probably be better to just have it always return a WebPusher object that is used with send rather than have instantiation differ in behavior so radically.